fvm/src/kernel/default.rs

```diff
@@ -790,6 +792,30 @@ where
     fn debug_enabled(&self) -> bool {
         self.call_manager.context().actor_debugging
     }
+
+    fn store_artifact(&self, name: &str, data: &[u8]) {
+        if let Ok(dir) = std::env::var(ENV_ARTIFACT_DIR) {
+            let mut dir = PathBuf::from(dir);
+
+            dir.push(self.call_manager.machine().machine_id().to_string());
+            dir.push(self.call_manager.origin().to_string());
+            dir.push(self.call_manager.nonce().to_string());
+            dir.push(self.actor_id.to_string());
+            dir.push(self.call_manager.invocation_count().to_string());
```

Comment:
nit: I think you can write something like.

```suggestion
            let dir: PathBuf = [
                self.call_manager.machine().machine_id().to_string(),
                self.call_manager.origin().to_string(),
                self.call_manager.nonce().to_string(),
                self.actor_id.to_string(),
                self.call_manager.invocation_count().to_string(),
            ].collect()
```